### PR TITLE
Draft: Feature/hot incident management

### PR DESCRIPTION
**Pull request by tobigremmer-dt**

### CODE REVIEW

File: user-skel/ansible/roles/hot-incident-management/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - { repo_target: "{{ hot_im_repo_name }}", repo_src: "{{ hot_im_source_repo }}", git_org: "{{ hot_im_org }}" }
+  when: new_repo_created is defined and new_repo_created == true
+
+# setEnv.sh should have DT URL, API and PAAS token before executing
+
+- name: Sed script file
+  shell: sed -i -e 's/\r$//' /home/ace/repos/hot/hot-im/setup.sh
+
+- name: Incident Mgmt Lab Setup
+  shell: . /home/ace/.bashrc && sh /home/ace/repos/hot/hot-im/setup.sh
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/hot-incident-management/tasks/main.yml]: Can we try to substitute this line with:

shell: "DYNATRACE_URL={{ dynatrace_tenant_url }} DYNATRACE_PAASTOKEN={{ dynatrace_paas_token }} DYNATRACE_APITOKEN={{ dynatrace_api_token }} sh /home/ace/repos/hot/hot-im/setup.sh"

File: user-skel/repos/hot-im/setup.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#Configure ITOP with some choice curl commands -- requires no parameters.
+echo "Configuring Itop..."
+chmod +x install_itop.sh
+sed -i 's/\r$//' install_itop.sh
+./install_itop.sh
+#TODO: re-configure on restart?
+
+# Run again to export ITOP_ENDPOINT
+source ./setEnv.sh
```
Comment (tobigremmer-dt) [inline user-skel/repos/hot-im/setup.sh]: I wouldn't run (and source) setEnv here as env variables are provided by Ansible.

File: user-skel/repos/hot-im/setup.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+#Run Monaco initial config
+echo "Running Dynatrace Monitoring as Code..."
+chmod +x executeMonaco.sh
+sed -i 's/\r$//' executeMonaco.sh
+source ./executeMonaco.sh
+
+#Install ITOP
+apt install docker.io -y
+sudo docker run -d -p 8000:80 --name=my-itop europe-west2-docker.pkg.dev/acetaskforceemea/images/itop:3.0.0-beta
```
Comment (kristofre) [inline user-skel/repos/hot-im/setup.sh]: @IgnacioGoldman why not deploy on Kubernetes?

File: user-skel/ansible/roles/hot-incident-management/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  register: ps
+
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/hot-incident-management/tasks/main.yml:68]: In addition, Ansible variables (such as dynatrace_tenant_url, dynatrace_api_token) can be subtituted and persisted in setEnv.sh for later.

File: user-skel/ansible/roles/hot-incident-management/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    git_domain: "{{ gitea_domain }}"
+    git_remote: "gitea"
+  loop:
+  - { repo_target: "{{ hot_im_repo_name }}", repo_src: "{{ hot_im_source_repo }}", git_org: "{{ hot_im_org }}" }
+  when: new_repo_created is defined and new_repo_created == true
+
+# setEnv.sh should have DT URL, API and PAAS token before executing
+
+- name: Sed script file
+  shell: sed -i -e 's/\r$//' /home/ace/repos/hot/hot-im/setup.sh
```
Comment (tobigremmer-dt) [inline user-skel/ansible/roles/hot-incident-management/tasks/main.yml]: Please don't hardcode "ace" user but use `{{ ace_box_user }}`, e.g. `/home/{{ ace_box_user }}/repos/hot/hot-im/setup.sh`

File: user-skel/repos/hot-im/setup.sh [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+source ./setEnv.sh
+
+#Run Monaco initial config
+echo "Running Dynatrace Monitoring as Code..."
+chmod +x executeMonaco.sh
+sed -i 's/\r$//' executeMonaco.sh
+source ./executeMonaco.sh
+
+#Install ITOP
+apt install docker.io -y
```
Comment (tobigremmer-dt) [inline user-skel/repos/hot-im/setup.sh]: Docker is installed by default. No need to re-install.

File: user-skel/repos/hot-im/setup.sh [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+#TODO: re-configure on restart?
+
+# Run again to export ITOP_ENDPOINT
+source ./setEnv.sh
+
+#########################
+# Install EasyTravel K8 #
+#########################
+cd k8-easytravel/scripts
+chmod +x create_easytravel.sh
```
Comment (tobigremmer-dt) [inline user-skel/repos/hot-im/setup.sh]: Optional (but would be pretty cool) if there was a dedicated role for Easytravel so other use cases could re-use it.

**Comment by tobigremmer-dt** [comment]
Repo has been migrated to https://github.com/dynatrace-ace/ace-box-ext-hot-incident-management

**Pull request closed without merging**